packages/beacon-node/src/metrics/validatorMonitor.ts

```diff
@@ -294,37 +293,43 @@ export function createValidatorMonitor(
       }
     },
 
-    registerValidatorStatuses(currentEpoch, status, index, balance) {
+    registerValidatorStatuses(currentEpoch, statuses, balances) {
       // Prevent registering status for the same epoch twice. processEpoch() may be ran more than once for the same epoch.
       if (currentEpoch <= lastRegisteredStatusEpoch) {
         return;
       }
       lastRegisteredStatusEpoch = currentEpoch;
 
-      switch (status) {
-        case "active":
-        case "active_exiting":
-        case "active_slashed":
-        case "active_ongoing":
-          metrics.validatorMonitor.validatorStatuses.inc({status: "active"});
-          break;
-        case "exited_unslashed":
-        case "withdrawal_done":
-          metrics.validatorMonitor.validatorStatuses.inc({status: "exited"});
-          break;
-        case "pending_initialized":
-        case "pending_queued":
-          metrics.validatorMonitor.validatorStatuses.inc({status: "pending"});
-          break;
-        case "withdrawal_possible":
-          metrics.validatorMonitor.validatorStatuses.inc({status: "withdrawable"});
-          break;
-        case "exited_slashed":
-          metrics.validatorMonitor.validatorStatuses.inc({status: "slashed"});
-          break;
-      }
+      metrics.validatorMonitor.validatorStatuses.reset();
+
+      for (const monitoredValidator of validators.values()) {
+        const index = monitoredValidator.index;
 
-      metrics.validatorMonitor.validatorBalances.set({index}, balance);
+        switch (statuses[index]) {
+          case "active":
+          case "active_exiting":
+          case "active_slashed":
+          case "active_ongoing":
+            metrics.validatorMonitor.validatorStatuses.inc({status: "active"});
```

Comment:
Much more efficient to compute this value first and call .set() once

Reply:
are you sure? It would require two iterations as opposed to one, as well as some extra space allocation for the incremented variables,  I don't think the call to `inc()` is more expensive than that.

Reply:
> I don't think the call to inc() is more expensive than that.

I would say so, please check prom-client source for the case of increasing with labels

Reply:
Good point, it could be more expensive depending on the amount of statuses given this [validateLabel](https://github.com/siimon/prom-client/blob/2c37e80c1bc14cd27eaeecd73cc36ad268856a2e/lib/validation.js#L17) check which iterates through all the labels.